### PR TITLE
Copy button for code blocks

### PR DESCRIPTION
**Pull request by pfitzseb**
image

Fixes https://github.com/JuliaDocs/Documenter.jl/issues/1055.

### CODE REVIEW

**Comment by fredrikekre** [comment]
I think it looks better with a copy icon like in https://github.com/JuliaDocs/Documenter.jl/pull/1454 instead of `Copy`

**Comment by pfitzseb** [comment]
Fair.

image
image

**Comment by fredrikekre** [comment]
Could the button be added only if javascript is enabled (i.e. at "runtime")? I did that here https://github.com/fredrikekre/fredrikekre.se/blob/master/_libs/clipboard.js for example. The way I did it there I don't need to do any unescaping, or include the code twice (once printed without colors as you do here IIUC). But maybe thats just because I used https://clipboardjs.com/ which handles that for me? Any reason not to use that?

**Comment by pfitzseb** [comment]
Sure, we can do that as well.

**Comment by fredrikekre** [comment]
Nice, add a changelog entry?

File: assets/html/scss/documenter/_elements.scss [hunk 1]
```diff
@@ -33,3 +33,35 @@ h1, h2, h3, h4, h5, h6 {
     display: none !important;
   }
 }
+
+pre {
+  position: relative;
+
+  .copy-button {
+    opacity: 0;
```
Comment (fredrikekre) [inline assets/html/scss/documenter/_elements.scss:41]: Perhaps make it visible without focus too? Something like
```suggestion
    opacity: 0.2;
```
looks pretty good IMO. Thoughts?

**Comment by fredrikekre** [comment]
Something buggy with the icons using the dark theme? I see this:
Screenshot 2022-01-13 at 22 32 54
for example (firefox, macos).

**Comment by fredrikekre** [comment]
Ah, looks like a CSS problem. In the dark theme `font-family` is overridden so it is missing `font-family: "Font Awesome 5 Free";`

**Comment by pfitzseb** [comment]
image

File: assets/html/scss/documenter/_elements.scss (hunk 1 above)
Comment (pfitzseb) [reply assets/html/scss/documenter/_elements.scss]: It does, yeah :)

**Pull request merged**